src/code-templates/libraries/async-local-storage/index.ts

```diff
@@ -0,0 +1,11 @@
+import { AsyncLocalStorage } from 'node:async_hooks';
+
+let currentContext: AsyncLocalStorage<unknown>;
+
+export function context<T = any>(): AsyncLocalStorage<T> {
```

Comment:
I'm asking myself what is the value in making this generic: It's useful when the instatiating code and the consuming code share the same type, for example when 'order-microservice' and 'logger' share some type but this is not the case - anyone might put here anything

As a golden rule, mostly in Practica, less is more. Generic is complication which _sometimes_ worth its price but is it indeed valuable here?

Reply:
Added generic in case the same code use the request context so it will have autocomplete

Let's say order service add user id to the context and some code later use that to fetch something

Maybe it shouldn't be allowed in the first place to make the function explicit of the used data

Reply:
Classic case of YAGNI, for **now** we don't have **specific** example where we **actually** need it === I would not introduce anything that is not needed